Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -246,6 +247,34 @@ public struct TargetSourcesBuilder {
         }
     }
 
+    private func diagnoseMissingResources(in resources: [Resource]) {
+        // Stop early if the valid resources array count is the same as the user defined list
+        if resources.count == self.target.resources.count {
+            return
+        }
+        
+        // Filter the package list of resources down to those which are not included in the valid resources array
+        let invalidResources = self.target.resources.filter { targetResource in
+            return resources.contains(where: { resource in
+                resource.path.relative(to: self.targetPath).pathString == targetResource.path
+            }) == false
+        }
+        
+        // If the list is empty then all were succesfully included so no error needs to be thrown
+        if invalidResources.isEmpty {
+            return
+        }
+        
+        var message = "target '\(self.target.name)' does not have resources at the following paths, they will be ignored\n"
+        
+        for resource in invalidResources {
+            let resourcePath = self.targetPath.appending(RelativePath(resource.path))
+            message += "    '\(resource.path)' (\(resourcePath))\n"
```

Comment:
This isn't a comment for this PR specifically, but I'm wondering if we should extend the diagnostics system to handle cases like this.  I think multi-line diagnostics won't necessarily be shown well in IDEs such as Xcode.  But this problem is not specific to this PR, so shouldn't block this from landing.  But I think we may need to do something across-the-board here to improve the diagnostics for IDEs, e.g. by providing a way to list associated paths in a semantically rich way rather than just as separate lines.

Reply:
Agreed - I believe when I was doing this I was loosely copying another muilti-line diagnostics I found so thought there was precedence to continue the pattern!